### PR TITLE
add support for mijn.host DNS challenge

### PR DESCRIPTION
**Pull request by shubhamku044**
## Summary by CodeRabbit

- **New Features**
	- Added support for the "mijn.host" DNS service, enhancing DNS challenge configuration options.
	- Introduced a new optional field for the Mijn Host API key in the configuration.

- **Bug Fixes**
	- Improved configuration handling for the new DNS provider to ensure proper API key integration.

- **Documentation**
	- Expanded the changelog to include version updates and support details for "mijn.host".
	- Updated documentation with example configurations for the new DNS provider.

- **Chores**
	- Updated configuration files and scripts to support the new DNS provider functionality.

### CODE REVIEW

**Comment by shubhamku044** [comment]
Add support for mijn.host dns challenge

**Review by agners — APPROVED** [review]

**Comment by agners** [comment]
This needs a newer ACME version. Let's update Certbot first before adding more DNS challenge options.

**Comment by shubhamku044** [comment]
okay, I have raised PR in certbot repo, waiting for the maintainers to merge it.

**Comment by agners** [comment]
Maybe I was a bit unclear: mijn certbot plug-in requires ACME 2.11.0, but we ship with 2.7.4 currently:

```
31.83     certbot-dns-mijn-host 0.0.3 depends on acme==2.11.0
```

Maybe you could ask `certbot-dns-mijn-host` to loosen that restriction, but I rather prefer to actually update the add-on to use the latest ACME 2.11.0. I am preparing a PR, hold on.

**Comment by shubhamku044** [comment]
by when can you merge this pr @agners

**Comment by agners** [comment]
@shubhamku044 with #3751 merged, you can now rebase/merge this PR and move forward.

File: letsencrypt/config.yaml
Comment (agners) [inline letsencrypt/config.yaml:103]: That got dropped with #3750. This probably came in through a merge conflict:

```suggestion
      dns-duckdns|dns-dynu|dns-gehirn|dns-godaddy|dns-google|\
```

**Review by agners — CHANGES_REQUESTED** [review]
Generally LGTM, thanks!

**Comment by agners** [comment]
Hm, seems there is still a merge conflict:

```
 #7 35.42 The conflict is caused by:
#7 35.42     The user requested cryptography==43.0.1
#7 35.42     certbot 2.11.0 depends on cryptography>=3.2.1
#7 35.42     certbot-dns-mijn-host 0.0.3 depends on cryptography==43.0.0
```

I guess the bump now bumped cryptography too high (to 43.0.1 :see_no_evil: ). Can the plug-in maybe loosen that restriction :thinking:

**Comment by agners** [comment]
FWIW, PR #3776 reverts Cryptography to 42 also for another reason, so you can rebase on this instead.

**Comment by shubhamku044** [comment]
hey @agners I have changed the version of cryptography, now it will work fine

**Comment by shubhamku044** [comment]
hi @agners

**Comment by kevinhagenaars** [comment]
When will this PR be merged into master? I was planning developing support for this provider, but then I saw there was this PR already with the functionality.

File: letsencrypt/CHANGELOG.md
Comment (agners) [inline letsencrypt/CHANGELOG.md:31]:
```suggestion
```

**Pull request merged**
